### PR TITLE
c7n_tencentcloud - cls, es, vpc, tcr - add resources

### PR DESCRIPTION
**Pull request by AndyChenGitHub**

### CODE REVIEW

File: tools/c7n_tencentcloud/tests/test_tc_tcr.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    def test_tcr(self):
+        policy = self.load_policy(
+            {
+                "name": "tcr-lifecycle-rule",
+                "resource": "tencentcloud.tcr",
+                "filters": [{"type": "lifecycle-rule", "state": False}]
+            }
+        )
+        resources = policy.run()
+        assert len(resources) > 0
```
Comment (kapilt) [inline tools/c7n_tencentcloud/tests/test_tc_tcr.py:19]: can we validate an attribute on the resource.

File: tools/c7n_tencentcloud/c7n_tencentcloud/resources/vpc.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    ie to find all vpcs with flows logs disabled we can do this
+
+    :example:
+    .. code-block:: yaml
+            policies:
+              - name: flow-mis-configured
+                resource: vpc
+                    filters:
+                      - type: flow-logs
+                        enabled: false
```
Comment (kapilt) [inline tools/c7n_tencentcloud/c7n_tencentcloud/resources/vpc.py:45]: same comment as per lifecycle-rule, users want to be able to filter on attributes of the flow log
https://www.tencentcloud.com/document/product/215/36908

File: tools/c7n_tencentcloud/tests/test_tc_vpc.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+                "filters": [
+                    {
+                        "type": "flow-logs",
+                        "enabled": False
+                    }
+                ]
+            }
+        )
+        resources = policy.run()
+        assert not resources
```
[the remark this answers is not in the document]
Comment (AndyChenGitHub) [reply tools/c7n_tencentcloud/tests/test_tc_vpc.py]: ok, done

File: tools/c7n_tencentcloud/tests/test_tc_tcr.py (hunk 1 above)
Comment (AndyChenGitHub) [reply tools/c7n_tencentcloud/tests/test_tc_tcr.py]: ok, done

File: tools/c7n_tencentcloud/c7n_tencentcloud/resources/tcr.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    """Lifecycle rule filtering
+
+    :Example:
+
+       policies:
+        - name: ecr-life
+          resource: tencentcloud.tcr
+          filters:
+            - type: lifecycle-rule
+              state: False
```
[the remark this answers is not in the document]
Comment (AndyChenGitHub) [reply tools/c7n_tencentcloud/c7n_tencentcloud/resources/tcr.py]: we made changes, field properties that match the api, can match key/value, If there are multiple rules, we only need to match one to pass, If there is a 'match' field, it needs to match all.

File: tools/c7n_tencentcloud/c7n_tencentcloud/resources/vpc.py (hunk 2 above)
Comment (AndyChenGitHub) [reply tools/c7n_tencentcloud/c7n_tencentcloud/resources/vpc.py]: same comment as per lifecycle-rule,  use ValueFilter to filter flow log attributes

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Pull request merged**